shell/common/api/electron_api_command_line.cc

```diff
@@ -12,18 +13,40 @@
 
 namespace {
 
-bool HasSwitch(const std::string& name) {
-  return base::CommandLine::ForCurrentProcess()->HasSwitch(name);
+// Chromium will hard CHECK if the switch name is not lowercase.
+bool IsSwitchNameValid(std::string_view switch_name) {
+  return base::ToLowerASCII(switch_name) == switch_name;
 }
 
-base::CommandLine::StringType GetSwitchValue(const std::string& name) {
-  return base::CommandLine::ForCurrentProcess()->GetSwitchValueNative(name);
+bool HasSwitch(gin_helper::ErrorThrower thrower,
+               const std::string& switch_string) {
+  if (!IsSwitchNameValid(switch_string)) {
+    thrower.ThrowError("Switch name must be lowercase");
+    return false;
+  }
+
+  return base::CommandLine::ForCurrentProcess()->HasSwitch(switch_string);
+}
+
+base::CommandLine::StringType GetSwitchValue(gin_helper::ErrorThrower thrower,
+                                             const std::string& switch_string) {
+  if (!IsSwitchNameValid(switch_string)) {
+    thrower.ThrowError("Switch name must be lowercase");
+    return base::CommandLine::StringType();
+  }
```

Comment:
Following up on the `CheckSwitchNameValid()` suggestion:

```suggestion
  if (const auto errmsg = CheckSwitchNameValid(switch_string)) {
    thrower.ThrowError(*errmsg);
    return {};
  }
```